server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -201,7 +226,7 @@ boolean isHeapUsageDominatedBySearch(List<SearchShardTask> searchShardTasks) {
     /**
      * Filters and returns the list of currently running SearchShardTasks.
      */
-    List<SearchShardTask> getSearchShardTasks() {
+    List<CancellableTask> getSearchShardTasks() {
```

Comment:
Those are 2 identical methods:

```
    <T extends CancellableTask & SearchBackpressureTask> List<CancellableTask> getTaskByType(Class<T> type) {
        return taskResourceTrackingService.getResourceAwareTasks()
            .values()
            .stream()
            .filter(type::isInstance)
            .map(type::cast)
            .collect(Collectors.toUnmodifiableList());
    }
```

Than you could use it to look by task type:

```
        List<CancellableTask> searchTasks = getTasksByType(SearchTask.class);
        List<CancellableTask> searchShardTasks = getTasksByType(SearchShardTask.class);

```

Reply:
+1 to avoid repetition.

Reply:
Updated the code to avoid repetition.